sunpy/map/header_helper.py

```diff
@@ -161,6 +160,13 @@ def make_fitswcs_header(data, coordinate,
          meta_wcs['PC2_1'], meta_wcs['PC2_2']) = (rotation_matrix[0, 0], rotation_matrix[0, 1],
                                                   rotation_matrix[1, 0], rotation_matrix[1, 1])
 
+    if hasattr(coordinate, 'rsun') and isinstance(coordinate.observer, frames.BaseCoordinateFrame):
+        observer = coordinate.observer.transform_to(
+            frames.HeliographicStonyhurst(obstime=coordinate.observer.obstime)
+        )
```

Comment:
```suggestion
```
These lines, as written, should never do anything.  The `coordinate.observer` attribute is already in `HeliographicStonyhurst` at the `coordinate.observer.obstime` obstime.